js/main.js

```diff
@@ -0,0 +1,221 @@
+    $navH1.textContent = "Question " + data.currentNumber;
+    createQuizQuestion();
+    console.log('Data model at start:', data)
+}
+
+//Reset Quiz?
+function resetQuiz(){
+    console.log('works');
+    var $quizContainer = document.querySelector('.quizContainer');
+    $quizContainer.remove();
+    $homePage.className = "container";
+    $quizPage.className = "container hidden"
+    $body.className = "";
+    $navH1.textContent = "Pokemon Quiz Game";
+    $navH2.className = "navH2 pokemon-font hidden"
+    data.currentNumber = 1;
+}
+
+function generateFour() {
+  data.currentFour = [];
+  for (var i = 0;; i++) {
+    var random = Math.floor(Math.random() * (151 - 1 + 1) + 1);
+    if (data.currentFour.includes(random)) {
+      continue;
+    } else {
+      data.currentFour.push(random);
+    }
+    if (data.currentFour.length === 4) {
+      break;
+    }
+  }
+  console.log('List of four Pokemon:', data.currentFour)
+}
+generateFour() //Array of 4 numbers, we want to reset this later.
+var shuffledFour = _.shuffle(data.currentFour);
+console.log("shuffledFour:", shuffledFour);
+
+function getPokemonName_1() {
+  return new Promise(function (resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', 'https://pokeapi.co/api/v2/pokemon/' + data.currentFour[0]);
+    xhr.onreadystatechange = function () {
+      if (this.readyState == 4) {
+        if (this.status == 200) {
+          resolve(this.responseText);
+        } else {
+          reject('Call 1 Failed');
+        }
+      }
+    };
+    xhr.send();
+  })
+}
+
+function getPokemonName_2() {
+  return new Promise(function (resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+      if (this.readyState == 4) {
+        if (this.status == 200) {
+          resolve(this.responseText)
+        } else {
+          reject('Call 2 Failed');
+        }
+      }
+    };
+    xhr.open('GET', 'https://pokeapi.co/api/v2/pokemon/' + data.currentFour[1]);
+    xhr.send();
+  })
+}
+
+function getPokemonName_3() {
+  return new Promise(function (resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+      if (this.readyState == 4) {
+        if (this.status == 200) {
+          resolve(this.responseText)
+        } else {
+          reject('Call 3 Failed');
+        }
+      }
+    };
+    xhr.open('GET', 'https://pokeapi.co/api/v2/pokemon/' + data.currentFour[2]);
+    xhr.send();
+  })
+}
+
+function getPokemonName_4() {
+  return new Promise(function (resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = function () {
+      if (this.readyState == 4) {
+        if (this.status == 200) {
+          resolve(this.responseText)
+        } else {
+          reject('Call 4 Failed');
+        }
+      }
+    };
+    xhr.open('GET', 'https://pokeapi.co/api/v2/pokemon/' + data.currentFour[3]);
+    xhr.send();
+  })
+}
```

Comment:
Is all of this to get names for multiple pokemon? You should only need to get data for the one pokemon that is the current question, right? If this is just to get names, I would make a variable containing all the pokemon names in an array, then just choose random indexes from that for the wrong answers, then only send a request for the pokemon you need the image for.